CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+project(facil.io C)
+
+find_package(Threads REQUIRED)
+
+set(facil.io_SOURCES
```

Comment:
@OwenDelahoy ,

This is good point. 

Let's leave the CMake as a list and I will edit the `makefile` so it recreates the `CMakeLists.txt` file  whenever I test things on my machine (I know, I will make sure it ignores `main.c`).

This will probably be the best way to automate the `CMakeLists.txt` updates and it will also solve the question of `target_include_directories`.